start-mailserver.sh

```diff
@@ -1,10 +1,55 @@
-#!/bin/sh
+#!/bin/bash
 
 die () {
   echo >&2 "$@"
   exit 1
 }
 
+# DKIM Setup
+mkdir -p /etc/opendkim/keys/$domainname
+if [ ! -f "/etc/opendkim/keys/$domainname/mail.private" ]; then
+  echo "Creating DKIM private key /etc/opendkim/keys/$domainname/mail.private"
+  pushd /etc/opendkim/keys/$domainname
+  opendkim-genkey --subdomains --domain=$domainname --selector=mail
+  popd
+  echo ""
+  echo "DKIM PUBLIC KEY ################################################################"
+  cat /etc/opendkim/keys/$domainname/mail.txt
+  echo "################################################################################"
+fi
+# Write to KeyTable if necessary
+if [ ! -f "/etc/opendkim/KeyTable" ]; then
+  echo "Creating DKIM KeyTable"
+  echo "mail._domainkey.$domainname $domainname:mail:/etc/opendkim/keys/$domainname/mail.private" > /etc/opendkim/KeyTable
+fi
+# Write to SigningTable if necessary
+if [ ! -f "/etc/opendkim/SigningTable" ]; then
+  echo "Creating DKIM SigningTable"
+  echo "*@$domainname mail._domainkey.$domainname" > /etc/opendkim/SigningTable
+fi
+echo "Changing permissions on /etc/opendkim"
+# chown entire directory
+chown -R opendkim:opendkim /etc/opendkim/
+# And make sure permissions are right
+chmod -R 0700 /etc/opendkim/keys/
+
+# Opendkim:
```

Comment:
These next lines are needed for debug but not on a production server.
(I have to clean the current debug by the way..)